pkg/query-service/app/opamp/logspipeline.go

```diff
@@ -0,0 +1,223 @@
+package opamp
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/knadh/koanf/parsers/yaml"
+	"github.com/open-telemetry/opamp-go/protobufs"
+	model "go.signoz.io/signoz/pkg/query-service/app/opamp/model"
+	"go.signoz.io/signoz/pkg/query-service/constants"
+	"go.uber.org/zap"
+)
+
+var lockLogsPipelineSpec sync.RWMutex
+
+func UpsertLogsParsingProcessor(ctx context.Context, parsingProcessors map[string]interface{}, parsingProcessorsNames []string, callback func(string, string, error)) (string, error) {
+	confHash := ""
+	if opAmpServer == nil {
+		return confHash, fmt.Errorf("opamp server is down, unable to push config to agent at this moment")
+	}
+
+	agents := opAmpServer.agents.GetAllAgents()
+	if len(agents) == 0 {
+		return confHash, fmt.Errorf("no agents available at the moment")
+	}
+
+	for _, agent := range agents {
+		config := agent.EffectiveConfig
+		c, err := yaml.Parser().Unmarshal([]byte(config))
+		if err != nil {
+			return confHash, err
+		}
```

Comment:
I am hoping there will be a PR for this as part of https://github.com/SigNoz/engineering-pod/issues/963 because I see this as a poorly thought out where the iterating over agents and updating it by `SendToAgent` multiple places. What was the manager designed to do?

Reply:
Yeah, I had a discussion with Amol, we can merge this into one. But It can be added as an enhancement to this PR as different things are happening while iterating over config for different use cases.